rand_pcg/src/pcg128.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2018 Developers of the Rand project.
+// Copyright 2017 Paul Dicker.
+// Copyright 2014-2017 Melissa O'Neill and PCG Project contributors
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! PCG random number generators
+
+// This is the default multiplier used by PCG for 64-bit state.
+const MULTIPLIER: u128 = 2549297995355413924u128 << 64 | 4865540595714422341;
+
+use core::{fmt};
+use rand_core::{RngCore, SeedableRng, Error, impls, le};
+
+/// A PCG random number generator (XSL 128/64 (MCG) variant).
+/// 
+/// Permuted Congruential Generator with 128-bit state, internal Multiplicative
+/// Congruential Generator, and 64-bit output via "xorshift low (bits),
+/// random rotation" output function.
+/// 
+/// This is equivalent to `pcg_engines::mcg_xsl_rr_128_64`. Note that we
```

Comment:
Would it be a good idea to link to that here? At very least saying what repo that came from might be a good idea.